tests/test-files/transform/MissingImportParens.elm

```diff
@@ -0,0 +1,14 @@
+module Main exposing (x)
+
+import Import.As1 as ImportAs1
+import Import.As2 as ImportAs2 exposing A
+import ImportDotDot exposing ..
+import ImportName exposing A
+import ImportNames exposing A, B, x, y
```

Comment:
I think there are multiline cases to consider.

What will happen with 

```
import ModuleName exposing A,

f = ()
```
(I guess in this specific example, parsing would fail... but specifically, I'm wondering are there any cases where a typo might cause the new parsing branches to end up successfully consuming a bunch of code it shouldn't, and lead to an unexpected formatting result?)

Do we want to allow things like
```
import ModuleName exposing
   A, B
```

Either we should be conservative, and only this feature when everything's on a single line, or make sure we have test cases for the multiline variants we can think of, and think through whether there are any cases that could lead to a bad result.

Reply:
Oh, you noted in the original post that you intentionally decided not to include line breaks... but `detailedListing` uses `commaSepBy1'`, which I think will consume comments, whitespace, and newlines immediately after a comma (maybe I'm wrong, though).  Did you try it to see what happens?

Reply:
`detailedListing` uses ` commaSep1'`.
https://github.com/avh4/elm-format/blob/master/parser/src/Parse/Helpers.hs#L246-L248
and `spaceySepBy1'`
https://github.com/avh4/elm-format/blob/master/parser/src/Parse/Helpers.hs#L236-L238
which uses `spaceySepBy1''`
https://github.com/avh4/elm-format/blob/master/parser/src/Parse/Helpers.hs#L214-L232

What's your idea to solve this?

Reply:
The example above with the line breaks should and is not working. I'm adding a test file for bad tests. But to be honest, I can't foresee the edge cases and their consequences.

Reply:
Okay, playing with this a bit more, if there is a trailing comma, then it will consume the next token, but any subsequent elm definition must start at the beginning of a line, so if an extra token was consumed, then it won't be the beginning of the line anymore and will fail to parse.  So I think this seems safe. 👍